main/src/ca/uwaterloo/flix/language/phase/Inliner.scala

```diff
@@ -34,19 +35,224 @@ import ca.uwaterloo.flix.util.Validation._
+      case Expression.Int64(_) => exp
+      case Expression.BigInt(_) => exp
+      case Expression.Str(_) => exp
+      case Expression.Var(_, _, _) => exp
+      case Expression.Ref(_, _, _) => exp
+      case Expression.Lambda(args, body, tpe, loc) =>
+        Expression.Lambda(args, visit(body), tpe, loc)
+      case Expression.Hook(_, _, _) => exp
+      case Expression.ApplyTail(sym, formals, actuals, tpe, loc) =>
+        Expression.ApplyTail(sym, formals, actuals.map(visit), tpe, loc)
+      case Expression.ApplyHook(hook, args, tpe, loc) =>
+        Expression.ApplyHook(hook, args.map(visit), tpe, loc)
+      case Expression.Apply(exp1, args, tpe, loc) =>
+        Expression.Apply(visit(exp1), args.map(visit), tpe, loc)
+      case Expression.Unary(op, exp1, tpe, loc) =>
+        Expression.Unary(op, visit(exp1), tpe, loc)
+      case Expression.Binary(op, exp1, exp2, tpe, loc) =>
+        Expression.Binary(op, visit(exp1), visit(exp2), tpe, loc)
+      case Expression.IfThenElse(exp1, exp2, exp3, tpe, loc) =>
+        Expression.IfThenElse(visit(exp1), visit(exp2), visit(exp3), tpe, loc)
+      case Expression.Let(sym, exp1, exp2, tpe, loc) =>
+        Expression.Let(sym, visit(exp1), visit(exp2), tpe, loc)
+      case Expression.LetRec(sym, exp1, exp2, tpe, loc) => None
+        Expression.LetRec(sym, visit(exp1), visit(exp2), tpe, loc)
+      case Expression.Is(sym, tag, exp1, loc) =>
+        Expression.Is(sym, tag, visit(exp1), loc)
+      case Expression.Tag(sym, tag, exp1, tpe, loc) =>
+        Expression.Tag(sym, tag, visit(exp1), tpe, loc)
+      case Expression.Untag(sym, tag, exp1, tpe, loc) =>
+        Expression.Untag(sym, tag, visit(exp1), tpe, loc)
+      case Expression.Index(base, offset, tpe, loc) =>
+        Expression.Index(visit(base), offset, tpe, loc)
+      case Expression.Tuple(elms, tpe, loc) =>
+        Expression.Tuple(elms.map(visit), tpe, loc)
+      case Expression.Existential(fparam, exp1, loc) =>
+        Expression.Existential(fparam, visit(exp1), loc)
+      case Expression.Universal(fparam, exp1, loc) =>
+        Expression.Universal(fparam, visit(exp1), loc)
+      case Expression.NativeConstructor(constructor, args, tpe, loc) =>
+        Expression.NativeConstructor(constructor, args.map(visit), tpe, loc)
+      case Expression.NativeField(_, _, _) => exp
+      case Expression.NativeMethod(method, args, tpe, loc) =>
+        Expression.NativeMethod(method, args.map(visit), tpe, loc)
+      case Expression.UserError(_, _) => exp
+      case Expression.MatchError(_, _) => exp
+      case Expression.SwitchError(_, _) => exp
+      /* Error */
+      case Expression.MkClosure(_, _, _, _) =>
+        throw InternalCompilerException(s"Unexpected expression $exp after lambda lifting")
+    }
+  }
 
-    root.toSuccess
+  /**
+    * Performs alpha-renaming of an expression
+    */
+  def renameAndSubstitute(exp: Expression, sub: Map[Symbol.VarSym, Symbol.VarSym])(implicit genSym: GenSym): Expression = {
```

Comment:
In idomatic Scala, if the body is a pattern match then you write:

```
def foo(bar: Baz): Int = bar match { 
```

without the additional brace before `=`.